$NetBSD: patch-da,v 1.1 2007/05/28 08:13:25 dan Exp $
#
# old_revision [f92dd754bf5c1e6eddc9c462b8d68691cfeb7f8b]
#
# patch "database.hh"
#  from [aa0ee865053a7c27df33ab9906ce681db1e9d6c8]
#    to [954a78c55234b282b7c047569cc5cc6ca9014341]
#
============================================================
--- database.hh	aa0ee865053a7c27df33ab9906ce681db1e9d6c8
+++ database.hh	954a78c55234b282b7c047569cc5cc6ca9014341
@@ -187,6 +187,7 @@ private:
   bool have_delayed_file(file_id const & id);
   void load_delayed_file(file_id const & id, file_data & dat);
   void cancel_delayed_file(file_id const & id);
+  void drop_or_cancel_file(file_id const & id);
   void schedule_delayed_file(file_id const & id, file_data const & dat);
 
   std::map<file_id, file_data> delayed_files;
@@ -215,6 +216,10 @@ private:
   bool delta_exists(std::string const & ident,
                     std::string const & table);
 
+  bool delta_exists(std::string const & ident,
+                    std::string const & base,
+                    std::string const & table);
+
   void get_file_or_manifest_base_unchecked(hexenc<id> const & new_id,
                                            data & dat,
                                            std::string const & table);
